packages/babel-traverse/src/path/conversion.ts

```diff
@@ -635,7 +635,7 @@ const getScopeInformationVisitor = mergeVisitors<{
 
       let curr = child.scope;
       do {
-        if (curr.hasOwnBinding("arguments")) {
+        if (curr.hasOwnBinding("arguments") && !curr.hasBinding("arguments")) {
```

Comment:
I think that rather than this, the proper fix would be to skip non-arrow functions in `scope.rename` if we are renaming `"arguments"`.

Reply:
Okay, I will do the same.